libraries/AP_FETtecOneWire/AP_FETtecOneWire.cpp

```diff
@@ -539,25 +541,52 @@ uint8_t AP_FETtecOneWire::init_escs()
     return _is.active_id;
 }
 
+/**
+    calculates crc tx error rate for incoming packages. It converts the CRC error counts into percentage
+    @param esc_id id of ESC, that the error is calculated for
+    @param esc_error_count the error count given by the esc
+    @param increment_only if this is set to 1 it only increases the message count and returns 0. If set to 1 it does not increment but gives back the error count.
+    @return the error in percent
+*/
+float AP_FETtecOneWire::calc_tx_crc_error_perc(uint8_t esc_id, uint16_t esc_error_count, uint8_t increment_only){

```

Comment:
Nope, the idea we discussed on the phone is to have this code in the ESC itself, it should transmit the error_rate*10000 over the wire as a uint16, and the ardupilot just calls:
'''
update_rpm(_telem_req_type-1, requested_telemetry[3]*100*2/pole_count.get(), requested_telemetry[5]);
'''